### PR TITLE
Refine WOQ Config

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change

[done] refine WoQ Config to RtnConfig, AwqConfig, TeqConfig, AutoRoundConfig, GPTQConfig, modify save and load function. @changwangss 
[done] cpu examples, uts, docs. @changwangss
[done] neuralspeed examples docs and uts. @changwangss
[done] xpu example and uts @PenghuiCheng 
[done] neuralchat example and uts @Spycsh 

Could you give me a review? @a32543254 @lvliang-intel @PenghuiCheng @hshen14 @kevinintel 
## Description

detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: intel_extension_for_transformers/transformers/__init__.py [hunk 1]
```diff
@@ -44,7 +44,11 @@
     MixedPrecisionConfig,
     BitsAndBytesConfig,
     SmoothQuantConfig,
-    WeightOnlyQuantConfig,
+    RtnConfig,
+    AwqConfig,
+    TeqConfig,
+    GPTQConfig,
```
Comment (a32543254) [inline intel_extension_for_transformers/transformers/__init__.py:50]: Why we use different config ？
I HF format, all method share one "quantization_config"
and use "quant_method": "gptq" to distinguish the methods

File: docs/weightonlyquant.md [hunk 2]
```diff
@@ -169,7 +171,7 @@ inputs = tokenizer(prompt, return_tensors="pt").input_ids.to(device)
 qmodel = AutoModelForCausalLM.from_pretrained(model_name, load_in_4bit=True, device_map="xpu", trust_remote_code=True)
 
 # optimize the model with ipex, it will improve performance.
-qmodel = ipex.optimize_transformers(qmodel, inplace=True, dtype=torch.float16, quantization_config={}, device="xpu")
+qmodel = ipex.optimize_transformers(qmodel, inplace=True, dtype=torch.float16, woq=True, device="xpu")
```
Comment (a32543254) [inline docs/weightonlyquant.md:174]: ipex.optimize_transformers is not support woq args anymore
and how about we hide it into generate api

File: intel_extension_for_transformers/transformers/__init__.py (hunk 1 above)
Comment (changwangss) [reply intel_extension_for_transformers/transformers/__init__.py:50]: all different have the save_pretrained method to save quantizate_config to config.json file, use different config will write different quant_method to config.json and quantize_config.json, it is the same behavoir with transformers.

File: docs/weightonlyquant.md [hunk 3]
```diff
@@ -33,19 +35,20 @@ As large language models (LLMs) become more prevalent, there is a growing need f
...
+Our motivation is improve CPU support for weight only quantization, since `bitsandbytes`, `auto-gptq`, `autoawq` only support CUDA GPU device. We have extended the `from_pretrained` function so that `quantization_config` can accept [`RtnConfig`](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/transformers/utils/config.py#L608), [`AwqConfig`](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/transformers/utils/config.py#L793), [`TeqConfig`](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/transformers/utils/config.py#L28), [`GPTQConfig`](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/transformers/utils/config.py#L855), [`AutoroundConfig`](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/transformers/utils/config.py#L912) to implement conversion on the CPU. We not only support PyTorch but also provide LLM Runtime backend based cpp programming language. Here are the example codes.
```
Comment (a32543254) [inline docs/weightonlyquant.md:42]: Our motivation is improve CPU support for weight only quantization,
>>
Our motivation is to improve CPU support for weight only quantization,

**Review by a32543254 — APPROVED** [review]
LGTM

File: examples/huggingface/pytorch/code-generation/quantization/run_generation.py [hunk 4]
```diff
@@ -93,57 +94,68 @@
...
 parser.add_argument(
-    "--gptq_percdamp",
+    "--damp_percent",
     type=float,
     default=0.01,
     help="Percent of the average Hessian diagonal to use for dampening.",
 )
 parser.add_argument(
-    "--gptq_block_size",
+    "--blocksize",
```
Comment (kevinintel) [inline examples/huggingface/pytorch/code-generation/quantization/run_generation.py:123]: HF is group_size?

File: examples/huggingface/pytorch/text-generation/quantization/run_tuning.sh [hunk 5]
```diff
@@ -157,64 +157,64 @@ function run_tuning {
...
         model_name_or_path="susnato/phi-1_5_dev"
         extra_cmd=$extra_cmd" --sq --alpha ${alpha}"
         extra_cmd=$extra_cmd" --output_dir ${tuned_checkpoint}"
-        extra_cmd=$extra_cmd" --trust_remote_code True"
+        extra_cmd=$extra_cmd" --trust_remote_code"
 	    pip install transformers==4.36.1
     elif [ "${topology}" = "llama2_7b_int4_gptq" ]; then
         model_name_or_path="meta-llama/Llama-2-7b-hf"
         extra_cmd=$extra_cmd" --woq --woq_weight_dtype int4_clip --woq_compute_dtype fp32"
         extra_cmd=$extra_cmd" --woq_algo "GPTQ" --gptq_actorder --gptq_block_size 128 --gptq_pad_max_length 2048 --gptq_use_max_length"
```
Comment (kevinintel) [inline examples/huggingface/pytorch/text-generation/quantization/run_tuning.sh:215]: --gptq_block_size

File: examples/huggingface/pytorch/code-generation/quantization/run_generation.py (hunk 4 above)
Comment (changwangss) [reply examples/huggingface/pytorch/code-generation/quantization/run_generation.py:123]: no, it is only for gptq needed. we have another args named "group_size" corresponding to HF "group_size"

Comment (hshen14) [reply examples/huggingface/pytorch/code-generation/quantization/run_generation.py:123]: what's the difference between group size and block size?

Comment (changwangss) [reply examples/huggingface/pytorch/code-generation/quantization/run_generation.py:123]: blocksize, use to set GPTQ sub weight matrix size
group_size, use to divide weight to calculate scale

**Review by hshen14 — APPROVED** [review]

File: docs/weightonlyquant.md (hunk 2 above)
Comment (hshen14) [reply docs/weightonlyquant.md:174]: shall we use ipex.llm.xxx?

File: docs/weightonlyquant.md (hunk 3 above)
Comment (changwangss) [reply docs/weightonlyquant.md:42]: improved.

File: docs/weightonlyquant.md (hunk 2 above)
Comment (changwangss) [reply docs/weightonlyquant.md:174]: @PenghuiCheng  Could you help to improve it if necessary?

**Review by lvliang-intel — APPROVED** [review]

File: docs/weightonlyquant.md (hunk 2 above)
Comment (changwangss) [reply docs/weightonlyquant.md:174]: align with Penghui, the model after ipex.optimize_transformers doesn't support save, we plan to improve it later, not in this PR.

**Comment by VincyZhang** [comment]
pylint is fixed in this PR: https://github.com/intel/intel-extension-for-transformers/pull/1371

**Pull request merged**